src/cloudant/result.py

```diff
@@ -360,10 +360,11 @@ def __getitem__(self, key):
 
         :returns: Rows data in JSON format
         """
-        if 'skip' in self.options or 'skip' in self._ref.keys():
+        refkeys = set(self._ref.keys())
+        if 'skip' in self.options or 'skip' in refkeys:
             msg = 'Cannot use skip parameter with QueryResult slicing.'
             raise CloudantArgumentError(msg)
-        if 'limit' in self.options or 'limit' in self._ref.keys():
+        if 'limit' in self.options or 'limit' in refkeys:
```

Comment:
Why the introduction of `refkeys` in this series of if statements?  This does not seem related to adding py3 support.